pkg/policy/compiler.go

```diff
@@ -83,23 +88,44 @@ func (c *compiler) Compile(policy *v1alpha1.AuthorizationPolicy) (PolicyFunc, fi
 			variables[variable.Name] = prog
 		}
 	}
-	var authorizations []cel.Program
+	var authorizations []authorizationPrograms
 	{
 		path := path.Child("authorizations")
 		for i, rule := range policy.Spec.Authorizations {
 			path := path.Index(i)
-			ast, issues := env.Compile(rule.Expression)
-			if err := issues.Err(); err != nil {
-				return nil, append(allErrs, field.Invalid(path.Child("expression"), rule.Expression, err.Error()))
-			}
-			if !ast.OutputType().IsExactType(envoy.CheckResponse) {
-				return nil, append(allErrs, field.Invalid(path.Child("expression"), rule.Expression, "rule output is expected to be of type envoy.service.auth.v3.CheckResponse"))
+			program := authorizationPrograms{}
+
+			if rule.Match != "" {
+				ast, issues := env.Compile(rule.Match)
+				if err := issues.Err(); err != nil {
+					return nil, append(allErrs, field.Invalid(path.Child("match"), rule.Match, err.Error()))
+				}
+				if !ast.OutputType().IsExactType(types.BoolType) {
+					return nil, append(allErrs, field.Invalid(path.Child("match"), rule.Match, "rule match output is expected to be of type bool"))
+				}
+				prog, err := env.Program(ast)
+				if err != nil {
+					return nil, append(allErrs, field.Invalid(path.Child("match"), rule.Match, err.Error()))
+				}
+				program.Match = prog
 			}
-			prog, err := env.Program(ast)
-			if err != nil {
-				return nil, append(allErrs, field.Invalid(path.Child("expression"), rule.Expression, err.Error()))
+
```

Comment:
We could probably extend the env with the envoy lib here so that it's not available to match programs